sympy/core/numbers.py

```diff
@@ -480,8 +483,13 @@ def mod_inverse(a, m):
         raise ValueError('inverse of %s (mod %s) does not exist' % (a, m))
     return c
 
+class ManagedABCMetaclass(ManagedProperties, ABCMeta):
+    pass
+
+class ManagedABCSingletonMetaclass(ManagedABCMetaclass, Singleton):
+    pass
 
-class Number(AtomicExpr):
+class Number(with_metaclass(ManagedABCMetaclass, AtomicExpr, numbers.Number)):
```

Comment:
Try if this passes the tests

```suggestion
class Number(with_metaclass(ManagedABCMetaclass, AtomicExpr, num.Number)):
```
Maybe the build errors are because python thinks you are referencing sympy.code.number.Numbers inside the definition of the same, so the module cannot have Number as it is not defined yet.